packages/cmrjs/tests/cmr-utils/test-cmr-utils.js

```diff
@@ -83,3 +91,69 @@ test('isCMRFile returns falsy if fileobject is invalid', (t) => {
   const fileObj = { bad: 'object' };
   t.falsy(isCMRFile(fileObj));
 });
+
+test('mapACNMTypeToCMRType returns a mapping', (t) => {
+  const mapCNMTypeToCMRType = cmrUtil.__get__('mapCNMTypeToCMRType');
+  t.is('EXTENDED METADATA', mapCNMTypeToCMRType('qa'));
+});
+
+test('mapACNMTypeToCMRType returns a default mapping if non CNM mapping specified', (t) => {
+  const mapCNMTypeToCMRType = cmrUtil.__get__('mapCNMTypeToCMRType');
+  t.is('GET DATA', mapCNMTypeToCMRType('NOTAREALVALUE'));
+});
+
+
+test.serial('updateEcho10XMLMetadata adds granule files correctly to OnlineAccessURLs/OnlineResources', async (t) => {
+  const uploadEchoSpy = sinon.spy(() => Promise.resolve);
+
+  const cmrXml = await fs.readFile('./tests/fixtures/cmrFileUpdateFixture.cmr.xml', 'utf8');
+  const cmrMetadata = await (promisify(xml2js.parseString))(cmrXml, xmlParseOptions);
+  const filesObject = await readJsonFixture('./tests/fixtures/filesObjectFixture.json');
+  const buckets = new BucketsConfig(await readJsonFixture('./tests/fixtures/buckets.json'));
+  const distEndpoint = 'https://distendpoint.com';
+
+  const updateEcho10XMLMetadata = cmrUtil.__get__('updateEcho10XMLMetadata');
+
+  const revertGenerateXml = cmrUtil.__set__('generateEcho10XMLString', () => 'testXmlString');
```

Comment:
What do these `revert` functions provide?

Reply:
They .... un-re-wire the methods.    It's probably not required, but doing it defensively felt like it made sense.

Reply:
@ifestus  and I had a conversation and he (correctly) noted this should be try/catch'd. 

64c7cddef